pandas/tests/frame/test_constructors.py

```diff
@@ -1424,6 +1424,25 @@ def test_from_dict_columns_parameter(self):
                 dict([("A", [1, 2]), ("B", [4, 5])]), columns=["one", "two"]
             )
 
+    @pytest.mark.parametrize(
+        "data_dict",
+        [
+            [{("a",): 1}, {("a",): 2}],
+            [OrderedDict([(("a",), 1), (("b",), 2)])],
+            [{("a", "b"): 1}],
+        ],
+    )
+    def test_constructor_from_dict_tuples(self, data_dict):
+        # GH 16769
+        df = DataFrame.from_dict(data_dict)
+
+        result = df.columns
+        all_columns = [key for sub_dict in data_dict for key in sub_dict.keys()]
```

Comment:
You can just parametrize the expected column values alongside the data instead of doing this

Reply:
got you - done